### ISSUE TITLE
Document how to diagnose compiler errors

### ISSUE THREAD

**Issue by mitzimorris** [stan-dev/stanc3#436]
We need documentation and procedures for end-users so that when existing models which were developed and tested via releases with the old stanc compiler are recompiled with the new stanc3 compiler, if there are any differences in performance or results, the users can figure out how to get help and/or how best to diagnose and document these behavoirs.

**Comment by seantalts** [comment]
Thanks for the issue! Could you give an example? Like would it be enough to show users how to try a previous version of the compiler? How do users currently test compiler changes?

**Comment by bob-carpenter** [comment]
> On Jan 20, 2020, at 4:11 PM, seantalts <notifications@github.com> wrote:
> 
> Thanks for the issue! Could you give an example? Like would it be enough to show users how to try a previous version of the compiler? How do users currently test compiler changes?

Not many of our users do SBC.  Typically, they find out when a model that worked in version N - 1 either gets different answers or gets slow in version N.

**Comment by rok-cesnovar** [comment]
After https://github.com/stan-dev/cmdstan/pull/799 is merged we can tell the users to add STANC2 to make/local if they suspect any issue. But that should probably be noted in the cmdstan readme or the release notes?

**Comment by wds15** [comment]
There must be big warnings with STANC2. This thing is not supported any more. New functionality and bug fixes will happen for STANC3 only in the future. So you can add a "deprecated" thing to STANC2 already now. At least this is how I would put it.

**Comment by rok-cesnovar** [comment]
Yes, Stanc2 should only be used to track down any bugs or performances regressions. I didnt mean it as "if you see any issue just use stanc2". My wording was a bit dubios, sorry.

**Comment by mitzimorris** [comment]
> After [stan-dev/cmdstan#799](https://github.com/stan-dev/cmdstan/pull/799) is merged we can tell the users to add STANC2 to make/local if they suspect any issue. But that should probably be noted in the cmdstan readme or the release notes?

agreed - this should go into CmdStan release notes.

**Comment by mitzimorris** [comment]
> There must be big warnings with STANC2. This thing is not supported any more. New functionality and bug fixes will happen for STANC3 only in the future.

that's not the issue here - the use case is a user who has a model that was developed for STANC2 and now it's been recompiled using STANC3 and now the model is either giving different answers or running slowly.  by using the old compiler but the same release version of stan_math, this will help determine whether the problem is the parser/generate or the math_lib or elsewhere.
